RATIONALE.md

```diff
@@ -112,6 +115,65 @@ Here are some examples:
 
 Besides security, this practice prevents other bugs and allows centralization of validation logic such as decoding Wasm.
 
+## API Design
+
+### Why are Go function signature parameters and results uint64?
+
+Go-defined function signatures include parameters and results expressed as a
```

Comment:
Isn't this only for the Call API? Host functions can naturally have any types other than u64

Reply:
good point. I guess I didn't write what I meant.. we have two apis to call functions from go..

I'll try to rewrite a bit as this is about invocation from go

Reply:
PTAL!